### PR TITLE
[SPARK-25989][ML] OneVsRestModel handle empty outputCols incorrectly

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
ignore empty output columns

## How was this patch tested?
added tests

### CODE REVIEW

**Comment by zhengruifeng** [comment]
friendly ping @srowen @jkbradley @MLnick

File: mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala [hunk 1]
```diff
@@ -219,14 +225,20 @@ final class OneVsRestModel private[ml] (
...
 
-      // output the index of the classifier with highest confidence as prediction
-      val labelUDF = udf { (rawPredictions: Vector) => rawPredictions.argmax.toDouble }
-
-      // output confidence as raw prediction, label and label metadata as prediction
-      aggregatedDataset
-        .withColumn(getRawPredictionCol, rawPredictionUDF(col(accColName)))
-        .withColumn(getPredictionCol, labelUDF(col(getRawPredictionCol)), labelMetadata)
-        .drop(accColName)
+      if (getPredictionCol != "") {
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala]: I guess I'm surprised these are both optional, in PredicitonModel too. But yeah consistency is good. However shouldn't this if clause be outside the "getRawPredictionCol = """ block? see ClassificationModel

Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala]: I implemented this in another way, classificationmodel update the output dataset, and I direct return the output in each if clause.
Then I update the to follow ClassificationModel, and update the outputColumns in each clauses. And `withColumns` is used to return the output columns.

File: mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala [hunk 2]
```diff
@@ -209,6 +215,9 @@ final class OneVsRestModel private[ml] (
       newDataset.unpersist()
     }
 
+    var outputColNames = Seq.empty[String]
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala]: Maybe 'predictionColumns' ? These aren't the only output columns. You could make this a mutable val too, but whatever.

**Review by srowen — COMMENTED** [review]
I think it's OK to implement it this way, but it's kind of inconsistent with Classifier. You could change that too but I don't feel strongly about it.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**